code/modules/client/client_procs.dm

```diff
@@ -1031,12 +1035,25 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 					movement_keys[key] = WEST
 				if("South")
 					movement_keys[key] = SOUTH
-				if("Say")
-					winset(src, "default-[REF(key)]", "parent=default;name=[key];command=say")
-				if("OOC")
-					winset(src, "default-[REF(key)]", "parent=default;name=[key];command=ooc")
-				if("Me")
-					winset(src, "default-[REF(key)]", "parent=default;name=[key];command=me")
+				if(SAY_CHAN)
+					var/say = create_window_message(SAY_CHAN)
+					winset(src, "default-[REF(key)]", "parent=default;name=[key];command=[say]")
+				if(RADIO_CHAN)
+					var/radio = create_window_message(RADIO_CHAN)
+					winset(src, "default-[REF(key)]", "parent=default;name=[key];command=[radio]")
+				if(ME_CHAN)
+					var/me = create_window_message(ME_CHAN)
+					winset(src, "default-[REF(key)]", "parent=default;name=[key];command=[me]")
+				if(OOC_CHAN)
+					var/ooc = create_window_message(OOC_CHAN)
+					winset(src, "default-[REF(key)]", "parent=default;name=[key];command=[ooc]")
+
+/** Creates a JSON encoded message to open TGUI modals properly */
+/client/proc/create_window_message(channel)
+	var/message = TGUI_CREATE_MESSAGE("open", list(
+		channel = channel,
+	))
+	return "\".output tgui_modal.browser:update [message]\""
```

Comment:
This proc name is inaccurate in regards to what it does.

Reply:
create_output_string ?

Reply:
tgui_modal_create_open_command